wgengine/wgcfg/parser.go

```diff
@@ -56,17 +57,13 @@ func parseEndpoint(s string) (host string, port uint16, err error) {
 	return host, uint16(uport), nil
 }
 
-func parseKeyHex(s string) (*wgkey.Key, error) {
-	k, err := hex.DecodeString(s)
-	if err != nil {
-		return nil, &ParseError{"Invalid key: " + err.Error(), s}
-	}
-	if len(k) != wgkey.Size {
-		return nil, &ParseError{"Keys must decode to exactly 32 bytes", s}
+// stringsCut is strings.Cut from proposed https://github.com/golang/go/issues/46336.
```

Comment:
It seems to me that we should add a Cut method to mem.RO.

Reply:
Makes sense, hopefully that won't block this PR tho